### PR TITLE
Gt 379 call to action custom color

### PR DESCRIPTION
**Pull request by ryancarlson**
Use control-color attribute to color the arrow.
Falls back to page.primary-color and manifest.primary-color in that order

### CODE REVIEW

**Comment by ryancarlson** [comment]
@frett

File: library/tract-renderer/src/main/java/org/cru/godtools/tract/model/CallToAction.java [hunk 1]
```diff
@@ -34,6 +36,9 @@
     @Nullable
     private Text mLabel;
 
+    @Nullable
+    private Integer mControlColor;
```
Comment (frett) [inline library/tract-renderer/src/main/java/org/cru/godtools/tract/model/CallToAction.java]: `@ColorInt`

it helps static analysis for lint checking

File: library/tract-renderer/src/main/java/org/cru/godtools/tract/model/CallToAction.java [hunk 2]
```diff
@@ -43,6 +48,9 @@
 
     @ColorInt
     private static int getArrowColor(@Nullable final CallToAction callToAction) {
+        if (callToAction != null && callToAction.mControlColor != null) {
+            return callToAction.mControlColor;
+        }
         return Styles.getPrimaryColor(callToAction != null ? callToAction.getPage() : null);
     }
```
Comment (frett) [inline library/tract-renderer/src/main/java/org/cru/godtools/tract/model/CallToAction.java]: I'd split this into 2 functions and rename to `getControlColor` now that we have an actual name for the attribute :)
```
private int getControlColor() {
    return mControlColor != null ? mControlColor : Styles.getPrimaryColor(getPage());
}

private int getControlColor(@Nullable final CallToAction callToAction) {
    return callToAction != null ? callToAction.getControlColor() : Styles.getPrimaryColor(null);
}
```

Comment (ryancarlson) [reply library/tract-renderer/src/main/java/org/cru/godtools/tract/model/CallToAction.java]: 👍

**Pull request merged**